tools/Makefile

```diff
@@ -30,10 +28,10 @@ aifc_decode_CFLAGS := -O2 # both runs and compiles faster than -O3
 aiff_extract_codebook_SOURCES := aiff_extract_codebook.c
 
 tabledesign_SOURCES := sdk-tools/tabledesign/codebook.c sdk-tools/tabledesign/estimate.c sdk-tools/tabledesign/print.c sdk-tools/tabledesign/tabledesign.c
-tabledesign_CFLAGS := -Wno-uninitialized -laudiofile -lstdc++
+tabledesign_CFLAGS := -laudiofile -lstdc++
 
 vadpcm_enc_SOURCES := sdk-tools/adpcm/vadpcm_enc.c sdk-tools/adpcm/vpredictor.c sdk-tools/adpcm/quant.c sdk-tools/adpcm/util.c sdk-tools/adpcm/vencode.c
-vadpcm_enc_CFLAGS := -Wno-unused-result -Wno-uninitialized -Wno-sign-compare -Wno-absolute-value
```

Comment:
Consider at the very least keeping `-Wno-unused-result`, because that's a pain in the ass and it doesn't help much in the context of where it appears.

Reply:
The way this PR fixes the warnings won't even require no-unused-result anymore anyway

Reply:
This has now truly been made essential as the warnings therein have been properly fixed.